tests/specialization_constants/specialization_constants_same_name_inter_link.h

```diff
@@ -0,0 +1,214 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for specialization constants with same name and internal
+//  linkage
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_SPEC_CONST_SAME_NAME_INTER_LINK_H
+#define __SYCLCTS_TESTS_SPEC_CONST_SAME_NAME_INTER_LINK_H
+
+#include "../common/common.h"
+#include "../common/once_per_unit.h"
+#include "specialization_constants_common.h"
+#include "specialization_constants_same_name_inter_link_helper.h"
+
+namespace specialization_constants_same_name_inter_link {
+using namespace sycl_cts;
+using namespace get_spec_const;
+
+template <typename T, int tu_num, typename via_kb>
+struct sc_sn_il_kernel;
+
+/** @brief Test suite for specialization constants with same name and internal
+ *         linkage
+ *  @tparam T Underlying data type, required to specify an actual instance of
+ *            the specialization constant
+ *  @tparam TestConfig helper class with default values depending on TU number
+ *  @tparam via_kb std::true(false)_type like helper type to separate tests
+ *                 with or without a kernel_bundle
+ */
+template <typename T, typename TestConfig, typename via_kb>
+class check_specialization_constants_same_name_inter_link_for_type {
+ public:
+  void operator()(util::logger &log, const std::string &type_name) try {
+    {
+      using kernel_name = sc_sn_il_kernel<T, TestConfig::tu, via_kb>;
+      constexpr auto &spec_const_il = sc_same_name_inter_link<T, via_kb>;
+      auto queue = util::get_cts_object::queue();
+      const sycl::context ctx = queue.get_context();
+      const sycl::device dev = queue.get_device();
+
+      if constexpr (via_kb::value) {
+        if (!dev.has(sycl::aspect::online_compiler)) {
+          once_per_unit::log(log, "Device does not support online compilation");
+          return;
+        }
+      }
+      sycl::range<1> range(1);
+      T def_value = T(value_helper<T>(0));
```

Comment:
```suggestion
      T def_value { value_helper<T> { 0 } };
```
and so on

Reply:
Change initialization to uniform in this commit: 7415160b25255d77d41c7e804eb1ec96223d56af